src/models/User.js

```diff
@@ -1,22 +1,66 @@
 export default (sequelize, DataTypes) => {
-  const User = sequelize.define('User', {
-    firstName: DataTypes.STRING,
-    lastName: DataTypes.STRING,
-    email: DataTypes.STRING,
-    password: DataTypes.STRING,
-  }, {
-    classMethods: {
-      associate: (models) => {
-        User.hasMany(models.Requests, {
-          onDelete: 'CASCADE',
-          foreignKey: 'UserId'
-        });
-        User.hasOne(models.Accomodations, {
-          foreignKey: 'UserId',
-          onDelete: 'CASCADE'
-        });
+  const User = sequelize.define(
+    'User',
+    {
+      firstName: DataTypes.STRING,
+      lastName: DataTypes.STRING,
+      email: {
+        type: DataTypes.STRING,
+        unique: true,
+        allowNull: false,
+        validate: {
+          isEmail: {
+            args: true,
+            msg: 'Email address is invalid.'
+          },
+          notNull: {
+            args: true,
+            msg: 'Email address field cannot be empty.'
+          },
+          async exists(value) {
+            const user = await User.findOne({
+              where: { email: value }
+            });
+
+            if (user) {
+              throw new Error('Email address already exists.');
```

Comment:
Due to security reasons, we must not tell the user that the email already exist. `User details already exist` is much better and safer

Reply:
Setting `unique` to true already caters for the code on line 20-30. I like the way you are able to show that we can use a custom function for validation.

Reply:
So you can set validation on the `unique` property and pass the error message i suggested above. i.e. `{ unique: { args: true, msg: 'User details already exist' } }`

Reply:
I tried it but it did not work. That is why I chose to use a custom validation for it.
I think I should delete line 9, 
```
{
  ...
  unique: true,
  ...
}
```
seeing that it is redundant.